core/src/main/scala/org/bitcoins/core/wallet/builder/RawTxBuilder.scala

```diff
@@ -0,0 +1,263 @@
+package org.bitcoins.core.wallet.builder
+
+import org.bitcoins.core.number.{Int32, UInt32}
+import org.bitcoins.core.protocol.script.{
+  CLTVScriptPubKey,
+  CSVScriptPubKey,
+  EmptyScriptSignature
+}
+import org.bitcoins.core.protocol.transaction.{
+  BaseTransaction,
+  EmptyWitness,
+  NonWitnessTransaction,
+  Transaction,
+  TransactionConstants,
+  TransactionInput,
+  TransactionOutput,
+  WitnessTransaction
+}
+import org.bitcoins.core.wallet.utxo.{
+  ConditionalInputInfo,
+  EmptyInputInfo,
+  InputInfo,
+  InputSigningInfo,
+  LockTimeInputInfo,
+  MultiSignatureInputInfo,
+  P2PKHInputInfo,
+  P2PKInputInfo,
+  P2PKWithTimeoutInputInfo,
+  P2SHInputInfo,
+  P2WPKHV0InputInfo,
+  P2WSHV0InputInfo,
+  UnassignedSegwitNativeInputInfo
+}
+
+import scala.annotation.tailrec
+import scala.collection.mutable
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+
+class RawTxBuilder {
+  private var version: Int32 = TransactionConstants.validLockVersion
+
+  private val inputsBuilder: mutable.Builder[
+    TransactionInput,
+    Vector[TransactionInput]] = Vector.newBuilder
+
+  private val outputsBuilder: mutable.Builder[
+    TransactionOutput,
+    Vector[TransactionOutput]] = Vector.newBuilder
+
+  private var lockTime: UInt32 = TransactionConstants.lockTime
+
+  private var finalizerOpt: Option[RawTxFinalizer] = None
+
+  def hasFinalizer: Boolean = finalizerOpt.isDefined
+
+  def setFinalizer(finalizer: RawTxFinalizer): this.type = {
+    finalizerOpt = Some(finalizer)
+    this
+  }
+
+  def result()(implicit ec: ExecutionContext): Future[Transaction] = {
+    finalizerOpt match {
+      case None =>
+        Future.failed(
+          new RuntimeException(
+            "Cannot call result with no RawTxFinalizer specified"))
+      case Some(finalizer) => result(finalizer)
+    }
+  }
+
+  def result(finalizer: RawTxFinalizer)(
```

Comment:
I think this method should just be `def result(): BaseTransaction` and then `RawTxFinalizer` can just take a `BaseTransaction` as input. This keeps the `Future` out of this API as well.